documentation/memory/step3/using_gc_traces.md

```diff
@@ -1,86 +1,229 @@
 # Tracing garbage collection
 
-There's a lot to learn about how the garbage collector works, but if you learn one thing it's that when GC is running, your code is not.
+This guide will go through the fundamentals of garbage collection traces.
 
-You may want to know how often and how long the garbage collection is running.
+By the end of this guide, you'll be able to:
+* Enable traces in your Node.js application
+* Interpret traces
+* Find memory leak source
 
-## Runnig with garbage collection traces
-You can see traces for garbage collection in console output of your process using the `--trace_gc` flag.
+There's probably a lot of stuff to learn about how the garbage collector works, but if you have to know one thing, it's that when GC is running, your code is not.
 
+You may want to know how often and long the garbage collection runs.
+
+## Setup
```

Comment:
I'd move it to the separated `.md` file. Maybe `exercise.md` ?

Reply:
(Sorry for the wait 🙈)

My attention here is to provide a similar experience to what we got in a tutorial. Moving it to a separate file will create overhead and degrade the reading experience.

Reply:
My concern is that it would need to be merged into the Node.js Guides along with the exercises package and I don't think it will work. My suggestion is to move it to a separate document and in the principal one (the one that gets merged recently in the nodejs guides) you include a mention to this exercise -- pointing to the diagnostics repository. What do you think?

Reply:
My idea was:
- in the guide
  - let the full readme with examples
  - add a link to the diag repository (to the `exercise` folder)
- in the diag repository
  - keep the `exercise` folder only

Sounds reasonable?

Reply:
Ok. I'll review the document as a set of examples then